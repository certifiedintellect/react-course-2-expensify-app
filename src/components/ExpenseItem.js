import React from 'react';
import {Link} from 'react-router-dom'

const ExpenseItem = ({id, description, amount, createdAt}) => (
    <div>
      <h3>{description} - <Link to={`/edit/${id}`}>Edit Expense</Link></h3>
      <p>{amount} - {createdAt}</p>
      
    </div>
)



export default ExpenseItem;