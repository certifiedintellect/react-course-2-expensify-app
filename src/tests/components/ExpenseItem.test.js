import React from 'react';
import {shallow} from 'enzyme';
import moment from 'moment';
import ExpenseItem from '../../components/ExpenseItem';

const expense = {
    id: '1',
    description: 'Gum',
    note: '',
    amount: 195,
    createdAt: moment(0)
}

test('should render expense item component on passinge expense', () => {
    const wrapper = shallow(<ExpenseItem {...expense}/>)
    expect(wrapper).toMatchSnapshot()
})