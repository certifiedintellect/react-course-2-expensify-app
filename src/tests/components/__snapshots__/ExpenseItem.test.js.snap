// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense item component on passinge expense 1`] = `
<div>
  <h3>
    Gum
     - 
    <Link
      to="/edit/1"
    >
      Edit Expense
    </Link>
  </h3>
  <p>
    195
     - 
    <Component />
  </p>
</div>
`;
